script.js

```diff
@@ -1,36 +1,115 @@
-  console.log(fileList);
-  for (const file of fileList) {
-    lib.appendImage(file);
+/* 
+    TO-DO:
+        Select filedetails and display it in the preview
+        Provide feature to remove selected files before upload
+        update the image array when the user remove a file.
+        Store the filedetails in the localstorage
+        on refresh display the image and provide a option delete to remove the image from dom and localstorage
+        Add remove feature to remove the image after reload.
+
+*/
+
+let fileToBeUploadedList = [];
+let image = [];
+let state = [];
+let savedImage = [];
+
+function toBase64(file) {
+  return new Promise((resolve, reject) => {
+    const reader = new FileReader();
+    reader.readAsDataURL(file);
+    reader.onload = () => resolve(reader.result);
+    reader.onerror = error => reject(error);
+  });
+}
+
+async function onFileSelect(event) {
+  const files = event.target.files;
+  for (const file of files) {
+    const isDuplicateFile = checkDuplicate(fileToBeUploadedList, file.name);
+    if (!isDuplicateFile) {
+      fileToBeUploadedList.push(file);
+      objtobase64 = await toBase64(file);
+      imgobj = {
+        name: file.name,
+        base64: objtobase64
+      };
+    }
+    const checkDuplicateinLocalStorage = checkDuplicate(image, file.name);
+    !checkDuplicateinLocalStorage ? image.push(imgobj) : null;
   }
-  fileList = [];
-  console.log(fileList);
+  helper.clearFileName();
+  helper.displayName(fileToBeUploadedList);
 }
 
-function checkDuplicate(file) {
-  return fileList.map(item => item.name).includes(file);
+function checkDuplicate(folder, file) {
+  return folder.map(item => item.name).includes(file);
 }
 
-function displayitem(files = fileDetails) {
-  for (const item of files) {
-    const isFileValid = checkDuplicate(item.name);
-    if (!isFileValid) {
-      lib.appendFileName(item.name);
-      fileList.push(item); //Single source of truth
-    }
+function storeInLocalStorage() {
+  console.log(image);
+  // displayImage.innerHTML = "";
+  helper.clearFileName();
+  try {
+    localStorage.setItem("FilesObject", JSON.stringify(image));
+  } catch (err) {
+    helper.exceededStorageCapcity;
+  }
+
+  console.log("Stored in LS");
+}
+
+function restoreSession() {
+  state = JSON.parse(localStorage.getItem("FilesObject"));
+  if (state) {
+    restoreImageOnReload();
   }
 }
 
-function getFileInfo(event) {
-  const data = event.dataTransfer;
-  files = data.files;
-  displayitem(files);
+function restoreImageOnReload() {
+  helper.displayHeader(state);
+  state.forEach(file => {
+    helper.restoreImage(file);
+    savedImage.push(file);
+  });
 }
 
-lib.addDisplayHandler(displayImage);
-lib.addDropHandler(getFileInfo);
+function removeImage(event) {
+  console.log(savedImage.length);
+  if (savedImage.length == 0) {
+    image.forEach(file => savedImage.push(file)); // storing the images from the localStorage.
+  }
+
+  helper.checkToRemoveHeader(savedImage);
```

Comment:
Remove check from inside the `checkToRemoveHeader` fn and move the condition here.